Task completed: GP-170. WeaponReplacrment

Kjusok opened the pull request:
https://lighteatersstudio.atlassian.net/jira/software/projects/GP/boards/1?selectedIssue=GP-170

1. Implemented a mechanism for changing weapons
2. Changing the UI when changing weapons in the HUD.
3. Not done - sound when changing weapons!

Kjusok committed:
Task completed: GP-170. WeaponReplacrment

1. Implemented a mechanism for changing weapons
2. Changing the UI when changing weapons in the HUD.
3. Not done - sound when changing weapons!

Kjusok committed:
Refactoring

Refactoring

frnzee commented on Assets/Scripts/Gameplay/Weapons/WeaponInitiator.cs:
```diff
@@ -0,0 +1,33 @@
+
+        private void Update()
+        {
+            if (Input.GetKeyDown(KeyCode.H))
```
Let's improve inputmap and use it for these actions instead of direct keycodes

ega-ega replied:
This is debug script. Current implementation normal. I marked script debug (rename)

EgorShaldovEpam committed:
- Refactoring
- Move files

EgorShaldovEpam committed:
- refactoring weapon debug script

EgorShaldovEpam committed:
- Refactoring Change weapon event(restore)
- Modified rifle prefab
- Refactored debug weapon initiator
- Added Disposed event to weapon
